apple/swift-docc
# Add support to diff DocC Archives

Bug/issue #, if applicable: 

## Summary

Adds a command line option for processing DocC Archives, producing a diff between two different versions of DocC Archives for the same framework or technology.

This diff contains a list of symbols added/removed between the two versions, and is stored in an output markdown file called `{FrameworkName}_ChangeLog.md` that will be written to the same directory as the input files.

The output markdown page can then be copy-pasted into the corresponding project and built as an article alongside the other documentation, serving as a starting point for writing change logs or updates pages.

To run this script, run the following command:

```
OVERVIEW: Generate a changelog with symbol diffs between documentation archives ('.doccarchive' directories).

USAGE: docc generate-changelog <initialDocCArchive> <newerDocCArchive> [--initial-archive-name <initial-archive-name>] [--newer-archive-name <newer-archive-name>] [--show-all <show-all>]

ARGUMENTS:
  <initialDocCArchive>    The path to the initial DocC Archive to be compared.
  <newerDocCArchive>      The path to the newer DocC Archive to be compared.

OPTIONS:
  --initial-archive-name <initial-archive-name>
                          The name of the initial DocC Archive version to be compared. (default: Version 1)
  --newer-archive-name <newer-archive-name>
                          The name of the newer DocC Archive version to be compared. (default: Version 2)
  --show-all <show-all>   Boolean value to indicate whether to produce a full symbol including all properties, methods, and overrides (default: false)
  -h, --help              Show help information.
```

## Dependencies

N/A

## Testing

Steps:
1. Run this command locally to diff two releases for a given framework, by passing in two local DocC Archives. 

    For example: 
    ```swift run docc process-archive diff-docc-archive "Release 1" /path/to/local/doccarchive/ "Release 2" /path/to/local/doccarchive2/``` 

2. Ensure that a new markdown file has been generated at the top level of the first DocC archive. This should be titled {FrameworkName}_ChangeLog.md (or NoFrameworkFound_ChangeLog.md if the framework name was not found).

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
